LeadKit/LeadKit/Extensions/UIImage/UIImage+Gradients.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  UIImage+Gradients.swift
+//  LeadKit
+//
+//  Created by Николай Ашанин on 28.09.16.
+//  Copyright © 2016 Touch Instinct. All rights reserved.
+//
+
+import UIKit
+
+public extension UIImage {
+
+    /**
+     creates a gradient color image.
+
+     - parameter gradientColors: An array of colors to use for the gradient.
+     - parameter size: Image size (defaults: 10x10)
+     */
+    public convenience init?(gradientColors: [UIColor],
+                      size: CGSize = CGSize(width: 10, height: 10)) {
+        UIGraphicsBeginImageContextWithOptions(size, false, 0)
+        let context = UIGraphicsGetCurrentContext()
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let colors = gradientColors.map {(color: UIColor) -> AnyObject? in return color.CGColor as AnyObject? } as NSArray
+        let gradient = CGGradientCreateWithColors(colorSpace, colors, nil)
+        CGContextDrawLinearGradient(context,
+                                    gradient,
+                                    CGPoint(x: 0, y: 0),
+                                    CGPoint(x: 0, y: size.height),
+                                    CGGradientDrawingOptions(rawValue: 0))
+        guard let cgImage = UIGraphicsGetImageFromCurrentImageContext().CGImage else {
+            return nil
+        }
+        self.init(CGImage: cgImage)
+        UIGraphicsEndImageContext()
+    }
+
+    /**
+     applies gradient color overlay to an image.
+
+     - parameter gradientColors: An array of colors to use for the gradient.
+     - parameter blendMode: The blending type to use.
+
+     - returns: A new image
+     */
+    public func applyGradientColors(gradientColors: [UIColor],
+                             blendMode: CGBlendMode = .Normal) -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(size, false, scale)
+        let context = UIGraphicsGetCurrentContext()
+        CGContextTranslateCTM(context, 0, size.height)
+        CGContextScaleCTM(context, 1.0, -1.0)
+        CGContextSetBlendMode(context, blendMode)
+        let rect = CGRect(x: 0, y: 0, width: size.width, height: size.height)
+        CGContextDrawImage(context, rect, CGImage)
+        // Create gradient
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let colors = gradientColors.map {(color: UIColor) -> AnyObject? in return color.CGColor as AnyObject? } as NSArray
```

Comment:
since array can't contain nil values I think you should use `flatMap` instead of `map`
```swift
gradientColors.flatMap { $0.CGColor } as CFArray
```